### DISCUSSION TITLE
Anyone got training with OCGOptV2 working successfully?

### DISCUSSION THREAD

**Discussion by uwidev** [67372a/LoRA_Easy_Training_Scripts#84] [General]
I had originally been using OCGOpt (not the v2) to train style gloras. I had recently found out about V2 and I've been having some troubles with training stability and good results.

I'm currently experimenting with various training configs.

Originally I used trained unet only at 3e-4, but because V2 doesn't use adaptive by default, it blows up the model.

I tried going to a normal LoRA at 1e-4 and results seem kind of okay. There's some instability and artifacting; like faces and limbs have a slight hard and malformed edge (though this was only at 500 steps instead of my usual 2000).

I also originally did `use_ramtorch`. I'm turning it off and doing trials to see if that affected results negatively. I recently completed a trial with `use_ramtorch_network` and that oddly led to nearly zero training. No idea what went wrong here.

If anyone is able to share some insights on working with this optimizer, I'd appreciate it.

**Comment by uwidev** [comment]
After some more trials, it seems a standard LoRA is usable at 1e-4.

But for generalized LoRA (gLoRA)... I'm still trying to figure that one out. At 5e-5 training seems to stabilized, but outputs are inconsistent. When using `use_scalar`, it seem to work better. Together it seems to produce the best results, but only about 70% of the time. The other 30% is disfigured. Pretty rough.

This one only does 5e-5 (no `use_scalar`), but it's still...
<img width="3840" height="1457" alt="grid" src="https://github.com/user-attachments/assets/c3b3f06b-01b9-448e-83a0-66fed3eed2f4" />

Tonight I'll run a trial with `use_scalar` and see how that goes. Maybe for now I'll cut the LR even lower and see how that goes.

Here is the results of using a LoRA at 1e-4. dim 8 alpha 8.
<img width="3844" height="1459" alt="grid_15" src="https://github.com/user-attachments/assets/0b43de1a-7b48-440f-9465-94584aaaeab4" />

**Comment by uwidev** [comment]
Good news. Maybe I just need to lower the LR. 2e-5 seems to be stable. Will experiment with this and will later post updates on how it goes. I have a lead now.
<img width="3820" height="1459" alt="grid_19" src="https://github.com/user-attachments/assets/f3788ffc-5773-4585-a9f5-c0e21d6dcae2" />

**Comment by uwidev** [comment]
I'll post any further breakthroughs if I find them, but 2e-5 seems to be a good starting point for having a stable and usable gLoRA. I forgot to set up validation testing, but at least after a trail I found that after 1k steps it starts to flatten out, compared to OCGOpt which needed around 2k.

Validation loss for some other artist I tried training.
<img width="1306" height="686" alt="image" src="https://github.com/user-attachments/assets/91aa783b-f0d7-49d0-8a93-2b005dbcac79" />

The 2k requirement on OGCOpt (v1) may have been due to adaptive. Earlier steps were definitely usable, but the validation loss just kept on dropping. I event kept training to like 4k and it only looked slightly overbaked, which was wild (batch 4).

Here's a 1e-5 (bottom) and a 2e-5 (top) trial I ran. The dataset I used was 32 images of [betanonbeet's](https://danbooru.donmai.us/posts?tags=betanonbeet) more recent works. They put emphasis on the eyelashes and adds heart highlights to it. Bright lighting, higher contrast, canvas textures.
<img width="3752" height="2833" alt="grid_02" src="https://github.com/user-attachments/assets/51deae34-f3bc-4472-85b1-57773b2b7546" />
